File: src/main/java/club/xiaojiawei/data/SpringData.java
package club.xiaojiawei.data;

import lombok.Getter;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Component;

/**
 * @author 肖嘉威
 * @date 2023/7/4 18:20
 * @msg 存储application.yaml里的值
 */
@Component
@Getter
public class SpringData {
    /**
     * 游戏日志配置路径
     */
    @Value("${game.log.configuration.path}")
    private String gameLogConfigurationPath;
    /**
     * 脚本配置文件路径
     */
    @Value("${script.configuration.file}")
    private String scriptConfigurationFile;
    @Value("${script.version}")
    private String version;
    @Value("${script.path}")
    private String scriptPath;
    @Value("${game.log.out.path}")
    private String gameLogPath;
    @Value("${game.log.out.file.screen}")
    private String screenLogName;
    @Value("${game.log.out.file.power}")
    private String powerLogName;
    @Value("${game.log.out.file.deck}")
    private String deckLogName;
    @Value("${spring.main.web-application-type}")
    private String webType;
    @Value("${server.port}")
    private String serverPort;
}

File: Hearthstone-Script/src/main/java/club/xiaojiawei/data/SpringData.kt
package club.xiaojiawei.data

import lombok.Getter
import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Component

/**
 * 存储application.yaml里的值
 * @author 肖嘉威
 * @date 2023/7/4 18:20
 */
@Component
class SpringData {
    /**
     * 游戏日志配置路径
     */
    @Value("\${game.log.configuration.path}")
    var gameLogConfigurationPath: String? = null

    /**
     * 脚本配置文件路径
     */
    @Value("\${script.configuration.file}")
    var scriptConfigurationFile: String? = null

    @Value("\${script.version}")
    var version: String? = null

    @Value("\${script.path}")
    var scriptPath: String? = null

    @Value("\${script.resource}")
    var resourcePath: String? = null

    @Value("\${game.log.out.path}")
    var gameLogPath: String? = null

    @Value("\${game.log.out.file.screen}")
    var screenLogName: String? = null

    @Value("\${game.log.out.file.power}")
    var powerLogName: String? = null

    @Value("\${game.log.out.file.deck}")
    var deckLogName: String? = null

    @Value("\${spring.main.web-application-type}")
    var webType: String? = null

    @Value("\${server.port}")
    var serverPort: String? = null

}
